src/gateways/mappers/BeaconRequestMapper.ts

```diff
@@ -0,0 +1,36 @@
+import { IBeacon } from "../../entities/IBeacon";
+import { IBeaconRequest } from "./IBeaconRequest";
+
+export interface IBeaconRequestMapper {
+  map: (beaconId: string, beacon: Partial<IBeacon>) => IBeaconRequest;
+}
+
+export class BeaconRequestMapper {
+  public map(beaconId: string, beacon: Partial<IBeacon>): IBeaconRequest {
+    const attributes: Record<string, string> = {};
+
+    if (beacon.hexId) attributes.hexId = beacon.hexId;
+    if (beacon.type) attributes.type = beacon.type;
+    if (beacon.protocolCode) attributes.protocolCode = beacon.protocolCode;
+    if (beacon.registeredDate)
+      attributes.registeredDate = beacon.registeredDate;
+    if (beacon.status) attributes.status = beacon.status;
+    if (beacon.manufacturer) attributes.manufacturer = beacon.manufacturer;
+    if (beacon.model) attributes.model = beacon.model;
+    if (beacon.manufacturerSerialNumber)
+      attributes.manufacturerSerialNumber = beacon.manufacturerSerialNumber;
+    if (beacon.chkCode) attributes.chkCode = beacon.chkCode;
+    if (beacon.batteryExpiryDate)
+      attributes.batteryExpiryDate = beacon.batteryExpiryDate;
+    if (beacon.lastServicedDate)
+      attributes.lastServicedDate = beacon.lastServicedDate;
+
+    return {
+      data: {
+        type: "beacon",
+        id: beaconId,
+        attributes: attributes,
+      },
+    };
+  }
+}
```

Comment:
There's an even uglier ES6 syntax for achieving this [here](https://medium.com/@mikeh91/conditionally-adding-keys-to-javascript-objects-using-spread-operators-and-short-circuit-evaluation-acf157488ede).
```typescript
const buildAnObjectFromAQuery = query => ({
  ...query.foo && { foo: query.foo },
  ...query.bar && { bar: query.bar },
});
```

I chose to go with simple inline if statements for readability and to make the mapping explicit.

Reply:
Fine :)